saw-core/src/SAWCore/Rewriter.hs

```diff
@@ -1043,16 +1053,24 @@ doHoistIfs sc ss hoistCache = go
            t' <- scApply sc f' x'
            return (t', conds1 ++ conds2)
 
-       goF _ (Lambda nm tp body) = goBinder scLambda nm tp body
-       goF _ (Pi nm tp body) = goBinder scPi nm tp body
+       goF _ (Lambda nm tp body) =
+          do let ec = EC nm tp
+             (body'', conds) <- go body
+             let (stuck, float) = List.partition (\(_,ecs) -> Set.member ec ecs) conds
 
-       goBinder close nm tp body = do
-           (ec, body') <- scOpenTerm sc nm tp 0 body
-           (body'', conds) <- go body'
-           let (stuck, float) = List.partition (\(_,ecs) -> Set.member ec ecs) conds
```

Comment:
I am a bit surprised to see `goBinder` be removed, especially since the `goF` cases for `Lambda` and `Pi` have so much repeated code. Is there a compelling reason not to de-duplicate the code shared in common?

Reply:
I had removed `goBinder` in an intermediate version when I was trying to convert Pi and Lambda binders in separate commits. Anyway, fixed in 22ee348e3509d59c10cb53bddf64f6678b703a3c.